### PR TITLE
Added form tests

### LINKED ISSUE #183: Get test coverage to 60%
# Feature
## Why is this feature being added?

Trying to get test coverage up to 90% incrementally. Currently at [Test Coverage](https://codeclimate.com/github/OperationCode/front-end/test_coverage). Let's move to 60%+ and I'll keep making tickets in 10% increments. You can check your progress locally via 

```
yarn jest --coverage
``` 

and looking at the top row.

## What should your feature do?
Add unit tests that move coverage to 60% or higher according to Code Climate report in PR

### PR DESCRIPTION
**Pull request by takethefake**
# Description of changes
Added a few more form tests

# Issue Resolved

Fixes #183

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit aee030077c65e90c883b08d81d5fa772af8a29e9

https://deploy-preview-184--operation-code-storybook.netlify.com

**Comment by takethefake** [comment]
Had to reimplement the selectbox, this has to be able to handle an own state, or switch to Downshift(https://github.com/paypal/downshift)

**Comment by kylemh** [comment]
heads up - just merged a PR to master right after u PRed. Sync locally 👍

**Comment by kylemh** [comment]
Resist the urge to refactor the form components too much going forward. Totally aware of how trash they are. We're going to be dispensing with them wholesale when implementing login/registration flow. I don't mind merging your refactors here, but I don't want your feelings hurt is later we elect to go a different direction. At the end of the day, your tests will help us implement past functionality safely while also being able to copy over the same validation logic tests 👍

File: components/_common_/Form/FormSelect/__tests__/FormSelect.test.js [hunk 1]
```diff
@@ -12,4 +13,66 @@ describe('FormSelect', () => {
...
+
+  test('should render with placeholder', () => {
+    createSnapshotTest(
+      <FormSelect
+        options={[{ label: 'test 1', value: 'TEST1' }, { label: 'test 2', value: 'TEST2' }]}
+      />,
+    );
+  });
+
+  test('should render with only with prompt', () => {
```
Comment (kylemh) [inline components/_common_/Form/FormSelect/__tests__/FormSelect.test.js]: `should render when passed a prompt and options`

File: components/_common_/Form/FormSelect/__tests__/FormSelect.test.js [hunk 2]
```diff
@@ -12,4 +13,66 @@ describe('FormSelect', () => {
       />,
     );
   });
+
+  test('should render with placeholder', () => {
```
Comment (kylemh) [inline components/_common_/Form/FormSelect/__tests__/FormSelect.test.js]: `should render when passing only required props`

File: components/_common_/Form/FormTextArea/__tests__/FormTextArea.test.js [hunk 3]
```diff
@@ -1,11 +1,26 @@
...
+import { mount } from 'enzyme';
 
 import FormTextArea from '../FormTextArea';
 
 describe('FormTextArea', () => {
   test('should render with just required props passed', () => {
     createSnapshotTest(<FormTextArea />);
   });
+
+  test('should render with placeholder', () => {
```
Comment (kylemh) [inline components/_common_/Form/FormTextArea/__tests__/FormTextArea.test.js]: `should render when passed a placeholder`

**Comment by takethefake** [comment]
I am currently not that sure how to test the SocialLogin best. will come back tomorrow with a solution that should bump the project up to 70 percent or above

**Comment by kylemh** [comment]
Social Login may be untouchable 😞 it relies on `react-router` which we used in the old repo.

If you want remove the code, I wouldn't be mad. We probably shouldn't have brought it in.

**Comment by kylemh** [comment]
I'm gonna go ahead and merge this. If you submit a PR simply deleting those components, I'll be okay with that.

File: components/_common_/Form/FormTextArea/__tests__/__snapshots__/FormTextArea.test.js.snap [hunk 4]
```diff
@@ -1,5 +1,15 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`FormTextArea should render when passed a placeholder 1`] = `
+<div>
+  <textarea
+    className="text_area"
+    onChange={[Function]}
+    placeholder=""
```
Comment (kylemh) [inline components/_common_/Form/FormTextArea/__tests__/__snapshots__/FormTextArea.test.js.snap]: 🤔 placeholder didn't actually get passed through

**Comment by takethefake** [comment]
got the 60 %!

File: components/SocialLogin/__tests__/SocialLogin.test.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { createMemoryHistory } from 'history';
```
Comment (kylemh) [inline components/SocialLogin/__tests__/SocialLogin.test.js]: woah! today I learned 🤔

File: components/_common_/Form/FormInput/FormInput.js [hunk 6]
```diff
@@ -11,7 +11,7 @@ class FormInput extends Component {
     placeholder: PropTypes.string,
     validateFunc: PropTypes.func,
     validationErrorMessage: PropTypes.string,
-    validationRegex: PropTypes.string,
+    validationRegex: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(RegExp)]),
```
Comment (kylemh) [inline components/_common_/Form/FormInput/FormInput.js]: 👏

File: components/_common_/Form/FormSelect/__tests__/FormSelect.test.js [hunk 7]
```diff
@@ -12,4 +13,66 @@ describe('FormSelect', () => {
       />,
     );
   });
+
+  test('should render when passing only required props', () => {
```
Comment (kylemh) [inline components/_common_/Form/FormSelect/__tests__/FormSelect.test.js]: duplicate test (line 9)

File: components/_common_/Form/FormTextArea/FormTextArea.js [hunk 8]
```diff
@@ -5,21 +5,22 @@ import styles from './FormTextArea.css';
 class FormTextArea extends Component {
   static propTypes = {
     onChange: PropTypes.func,
-    placeHolder: PropTypes.string,
+    placeholder: PropTypes.string,
```
Comment (kylemh) [inline components/_common_/Form/FormTextArea/FormTextArea.js]: 😅 good fix - thanks

File: test-utils/mocks/axios.js [hunk 9]
```diff
@@ -0,0 +1,3 @@
+import mockAxios from 'jest-mock-axios';
```
Comment (kylemh) [inline test-utils/mocks/axios.js]: 💃

**Pull request merged**